externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -514,4 +515,41 @@ def Tcp_Atan2Op : Tcp_BinaryElementwiseOp<"atan2", [SameOperandsAndResultElement
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_CastOp:  Tcp_Op<"cast", [Pure, Elementwise, SameOperandsAndResultShape]> {
+
+  let summary = "TCP Cast operation";
+
+  let description = [{
+    Performs elementwise type conversion.
+
+    Supported torch/mlir element type:
+      INT: ui8,  si8,  i8, 
+           ui16, si16, i16,
+           ui32, si32, i32,
+           ui64, si64, i64, 
+           i1 (Boolean)
+      FP: f16, bf16, f32, f64
+
+    Supported Conversion Mode:
+      FP -> FP
+      INT -> FP
+      FP -> INT
+      INT -> INT
```

Comment:
There are quite a few corner cases here, e.g. what happens if the conversion overflows the destination type or what happens if a negative number is converted to an unsigned destination type. Consider explicitly speccing them at some point.

Reply:
Thats a good point. @zezhang Can you add a TODO here to handle other corner cases?